app/controllers/my_facilities_controller.rb

```diff
@@ -36,39 +36,18 @@ def index
   end
 
   def bp_controlled
-    @facilities = filter_facilities
-
-    @data_for_facility = {}
-
-    @facilities.each do |facility|
-      @data_for_facility[facility.name] = Reports::RegionService.new(region: facility, period: @period).call
-    end
-
-    @facilities_by_size = @facilities.group_by { |facility| facility.facility_size }
+    facilities = filter_facilities
+    process_facility_stats(facilities, "controlled_patients")
   end
 
   def bp_not_controlled
-    @facilities = filter_facilities
-
-    @data_for_facility = {}
-
-    @facilities.each do |facility|
-      @data_for_facility[facility.name] = Reports::RegionService.new(region: facility, period: @period).call
-    end
-
-    @facilities_by_size = @facilities.group_by { |facility| facility.facility_size }
+    facilities = filter_facilities
+    process_facility_stats(facilities, "uncontrolled_patients")
```

Comment:
The numerator arg should probably be a symbol - `:uncontrolled_patients` for example.